File: Examples/Deprecated/Export-Device-Inventory/Perl/exportInventory.pl
use LWP::UserAgent;
use HTTP::Cookies;
use JSON::RPC::Client2;
use Data::Dumper;
use POSIX qw(strftime);
use Text::CSV;

### Initial authentication
my $cookie_jar = HTTP::Cookies->new( ignore_discard => 1 );
my $ua = LWP::UserAgent->new(
    cookie_jar => $cookie_jar,
    ssl_opts   => { SSL_use_cert => 0, verify_hostname => 0 }
);

my $output_file = ($#ARGV == -1) ? "InventoryReport" . (strftime "%y%m%d-%H%M%S", localtime) . ".csv" : $ARGV[0];

my $netld = JSON::RPC::Client2->new("https://localhost/rest?j_username=admin&j_password=password", $ua);

my $pageData = $netld->call("Inventory.search", {
        network  => 'Default',
        scheme   => 'ipAddress',
        query    => '',
        pageData => {
            offset   => 0,
            pageSize => 500,
        },
        sortColumn => 'ipAddress',
        descending => 0
    }
);

do {
	my $csv = Text::CSV->new ({binary => 1, eol => "\n"}) or die "Cannoot use CSV: " .Text::CSV->error_diag(); 
	open $fh, ">>", $output_file or die "$output_file: $!"; 
	foreach $device (@{$pageData->{devices}})
	{
		$csv->print ($fh, [$device->{backupStatus}, $device->{ipAddress}, 
						$device->{hostname}, $device->{hardwareVendor},              
						$device->{model}, $device->{deviceType}, 
						$device->{serialNumber}, $device->{adapterId}, 
						$device->{osVersion}, $device->{softwareVendor}, 
						$device->{backupElapsed}, $device->{memoSummary},     
						$device->{custom1}, $device->{custom2}, $device->{custom3}, 
						$device->{custom4}, $device->{custom5}]);
	}
	close $fh;
	
	$pageData->{offset} += $pageData->{pageSize}
}
until ($pageData->{total} <= $pageData->{offset} + $pageData->{pageSize});

$netld->call("Security.logoutCurrentUser");

1;

File: Examples/Deprecated/Export-Device-Inventory/Ruby/exportInventory.rb
#!/usr/bin/env ruby -W0

require 'jimson'
require 'csv'

OpenSSL::SSL::VERIFY_PEER = OpenSSL::SSL::VERIFY_NONE

if ARGV.empty?
  outputFile = "InventoryReport" + DateTime.parse(Time.now.to_s).strftime("%y%m%d-%H%M%S") + ".csv"
else
  outputFile = ARGV[0]
end

netld = Jimson::Client.new("https://10.0.40.50/rest?j_username=admin&j_password=password")

pageData = { 'offset' => 0, 'pageSize' => 100, 'total' => 500 }

until pageData['offset'] + pageData['pageSize'] >= pageData['total'] do   
  pageData = netld['Inventory.search', 'Default', 'ipAddress', "", pageData, 'ipAddress', false]
  
  CSV.open(outputFile, "wb") do |csv|
    for device in pageData['devices']
      csv << [device['backupStatus'], device['ipAddress'], device['hostname'], device['hardwareVendor'],                
              device['model'], device['deviceType'], device['serialNumber'],device['adapterId'],                 
              device['osVersion'], device['softwareVendor'], device['backupElapsed'], device['memoSummary'],                
              device['custom1'], device['custom2'], device['custom3'], device['custom4'], 
              device['custom5']]
    end
  end
  pageData['offset'] += pageData['pageSize']
end

netld['Security.logoutCurrentUser']
